src/safir/testing/kubernetes.py

```diff
@@ -742,6 +824,426 @@ async def list_namespaced_event(
+            How long to return events for before exiting when performing a
+            watch.
+        watch
+            Whether to act as a watch.
+        _preload_content
+            Verified to be `False` when performing a watch.
+        _request_timeout
+            Ignored, accepted for compatibility with the watch API.
+
+        Returns
+        -------
+        kubernetes_asyncio.client.V1IngressList or unittest.mock.Mock
+            List of ingresss in that namespace, when not called as a watch. If
+            called as a watch, returns a mock ``aiohttp.Response`` with a
+            ``readline`` metehod that yields the events.
+
+        Raises
+        ------
+        AssertionError
+            Some other ``field_selector`` was provided.
+        kubernetes_asyncio.client.ApiException
+            Raised with 404 status if the namespace does not exist.
+        """
+        self._maybe_error("list_namespaced_ingress", namespace, field_selector)
+        if namespace not in self._objects:
+            msg = f"Namespace {namespace} not found"
+            raise ApiException(status=404, reason=msg)
+        if not watch:
+            if field_selector:
+                match = re.match(r"metadata\.name=(.*)$", field_selector)
+                assert match and match.group(1)
+                try:
+                    ingress = self._get_object(
+                        namespace, "Ingress", match.group(1)
+                    )
+                    return V1IngressList(kind="Ingress", items=[ingress])
+                except ApiException:
+                    return V1IngressList(kind="Ingress", items=[])
+            else:
+                ingresss = []
+                if "Ingress" in self._objects[namespace]:
+                    for obj in self._objects[namespace]["Ingress"].values():
+                        ingresss.append(obj)
+                return V1IngressList(kind="Ingress", items=ingresss)
+
+        # All watches must not preload content since we're returning raw JSON.
+        # This is done by the Kubernetes API Watch object.
+        assert not _preload_content
+
+        # Return the mock response expected by the Kubernetes API.
+        stream = self._event_streams[namespace]["Ingress"]
+        return stream.build_watch_response(
+            resource_version,
+            timeout_seconds,
+            field_selector=field_selector,
+            label_selector=label_selector,
+        )
+
+    async def patch_namespaced_ingress_status(
+        self, name: str, namespace: str, body: list[dict[str, Any]]
+    ) -> V1Secret:
+        """Patch the status of an ingress object.
+
+        Parameters
+        ----------
+        name
+            Name of ingress object.
+        namespace
+            Namespace of secret object.
+        body
+            Patches to apply. Only patches with ``op`` of ``replace`` are
+            supported, and only with ``path`` of
+            ``/status/load_balancer/ingress``.
```

Comment:
This should be ``/status/loadBalancer/ingress`` since patches use the Kubernetes API field names, not the Python renamings of them.